src/sql/src/session/vars.rs

```diff
@@ -860,6 +860,15 @@ const PG_REPLICATION_TCP_USER_TIMEOUT: ServerVar<Duration> = ServerVar {
     internal: true,
 };
 
+/// Sets the `statement_timeout` value to use during the snapshotting phase of
+/// PG sources.
+const PG_SNAPSHOT_STATEMENT_TIMEOUT: ServerVar<Duration> = ServerVar {
+    name: UncasedStr::new("pg_snapshot_statement_timeout"),
+    value: &mz_postgres_util::DEFAULT_SNAPSHOT_STATEMENT_TIMEOUT,
+    description: "Sets the `statement_timeout` value to use during the snapshotting phase of PG sources (Materialize)",
+    internal: true,
+};
```

Comment:
Will you do me a favor and make sure this works properly for durations of `0`? I think it gets mapped to `usize::MAX` which might not work? But also you wrote a test and that didn't choke. So IDK.

Slack context about the weirdness of duration configuration parameters: https://materializeinc.slack.com/archives/C02FWJ94HME/p1691674999794559